monitoring/check_collectd_mlab.py

```diff
@@ -0,0 +1,461 @@
+          'Read an empty message from collectd socket.')
+
+
+def assert_collectd_vsys_setup():
+  """Asserts that vsys configuration is complete for mlab_utility.
+
+  Raises:
+    CriticalError if an error occurs.
+  """
+  # Is the vsys backend script installed?
+  if not os.path.exists(VSYSPATH_BACKEND):
+    raise MissingVsysBackendCriticalError(
+        'The vsys backend script %s is missing!' % VSYSPATH_BACKEND)
+
+  # Is the vsys frontend FIFO in the slice context?
+  if not os.path.exists(VSYSPATH_SLICE):
+    raise MissingVsysFrontendCriticalError(
+        'The vsys frontend fifo %s is missing in slice!' % VSYSPATH_SLICE)
+
+  # Is mlab_utility in the vsys acl for the backend script?
+  try:
+    acl = open(VSYSPATH_ACL, 'r').read().strip()
+  except IOError:
+    raise MissingVsysAclCriticalError(
+        'Failed to read the vsys ACL: %s' % VSYSPATH_ACL)
+
+  if SLICENAME not in acl:
+    raise MissingSliceFromVsysAclCriticalError(
+        'Slice name %s is missing from ACL: %s' % (SLICENAME, VSYSPATH_ACL))
+
+
+def run_collectd_nagios(host, metric, value, warning, critical):
+  """Runs collectd-nagios using given parameters as arguments.
+
+  Please see the collectd-nagios man page for documentation on the format of
+  'warning' and 'critical' thresholds.
+
+  Args:
+    host: str, experiment hostname.
+    metric: str, raw metric path. (e.g. network/if_octets-ipv6)
+    value: str, name of value within metric. (e.g. 'value', 'rx')
+    warning: str, a collectd-nagios warning threshold. (e.g. '0:20')
+    critical: str, a collectd-nagios critical threshold. (e.g. '0:30')
+  Returns:
+    int, exit code from collectd-nagios. Because this is a nagios plugin, these
+        are valid nagios exit states.
+  """
+  env = os.environ.copy()
+  env['LD_LIBRARY_PATH'] = '%s:%s' % (
+      LD_LIBRARY_PATH, env.get('LD_LIBRARY_PATH', ''))
+  cmd = ('{collectd_nagios} -s {unixsock} -H {host} -n {metric} -d {value} '
+         '-w {warning} -c {critical}')
+  cmd = cmd.format(
+      collectd_nagios=COLLECTD_NAGIOS, unixsock=COLLECTD_UNIXSOCK, host=host,
+      metric=metric, value=value, warning=warning, critical=critical)
+  child = subprocess.Popen(
+      cmd.split(), stdout=subprocess.PIPE, stderr=subprocess.PIPE, env=env)
+  return child.wait()
+
+
+def assert_collectd_nagios_levels():
+  """Asserts actual values from collectd using collectd-nagios.
+
+  Asserts that storage quota is sufficient, metric collection is fast enough,
+  that CPU load is low, and memory usage is reasonable.
+
+  Raises:
+    NagiosStateError, if an error occurs.
+  """
+  # TODO: Make warning & critical thresholds configurable.
+
+  # Is utility slice quota ok?
+  exit_code = run_collectd_nagios(
+      'utility.mlab.' + HOSTNAME, 'storage/vs_quota_bytes-quota',
+      'used', _mb_to_bytes(8000), _mb_to_bytes(9000))
+  if exit_code != 0:
+    raise NagiosStateError('Storage quota usage is too high',
+                           status_code=exit_code)
```

Comment:
Why use keyword arguments here? I feel like 

```
raise NagiosStateError('Storage quota usage is too high', exit_code)
```
is pretty clear (and has the benefit of fitting on one line)